functionary/schema.py

```diff
@@ -9,7 +9,262 @@
+    """Append a new parameter with comment to the info_list
+
+    Args:
+        info_lines (List[str]): current info_list
+        param_declaration (str): param: type
+        comment_info (Optional[str]): information of comment
+        depth (int): level of nested param
+    """
+    offset = ""
+    if depth >= 1:
+        offset = "".join(["    " for _ in range(depth)])
+    if comment_info is not None:
+        # if depth == 0:  # format: //comment\nparam: type
+        info_list.append(f"{offset}{comment_info}")
+        info_list.append(f"{offset}{param_declaration}")
+    # else:  # format: param: type  // comment
+    #     info_list.append(f"{offset}{param_declaration}    {comment_info}")
+    else:
+        info_list.append(f"{offset}{param_declaration}")
+
+
+def get_enum_option_str(enum_options: List) -> str:
+    """get enum option separated by: "|"
+
+    Args:
+        enum_options (List): list of options
+
+    Returns:
+        _type_: concatenation of options separated by "|"
+    """
+    # if each option is string --> add quote
+    return " | ".join([f'"{v}"' if type(v) is str else str(v) for v in enum_options])
+
+
+def get_array_typescript(param_name: Optional[str], param_dic: dict, depth: int = 0) -> str:
+    """recursive implementation for generating type script of array
+
+    Args:
+        param_name (Optional[str]): name of param, optional
+        param_dic (dict): param_dic
+        depth (int, optional): nested level. Defaults to 0.
+
+    Returns:
+        _type_: typescript of array
+    """
+    offset = ""
+    if depth >= 1:
+        offset = "".join(["    " for _ in range(depth)])
+    items_info = param_dic.get("items", {})
+
+    if len(items_info) == 0:
+        if param_name is not None:
+            return f"{offset}{param_name}: []"
+        else:
+            return "[]"
+    array_type = get_param_type(items_info)
+    if array_type == "object":
+        info_lines = []
+        child_lines = get_parameter_typescript(
+            items_info.get("properties", {}), items_info.get("required", []), depth + 1
+        )
+        # if comment_info is not None:
+        #    info_lines.append(f"{offset}{comment_info}")
+        if param_name is not None:
+            info_lines.append(f"{offset}{param_name}" + ": {")
+        else:
+            info_lines.append(f"{offset}" + "{")
+        info_lines.extend(child_lines)
+        info_lines.append(f"{offset}" + "}[]")
+        return "\n".join(info_lines)
+
+    elif array_type == "array":
+        item_info = get_array_typescript(None, items_info, depth + 1)
+        if param_name is None:
+            return f"{item_info}[]"
+        return f"{offset}{param_name}: {item_info.strip()}[]"
+
+    else:
+        if "enum" in items_info:
+            item_type = get_enum_option_str(items_info["enum"])
+            if param_name is None:
+                return f"({item_type})[]"
```

Comment:
Oh this is for handling array so we need to add [], because array is quite complex so I wrote a separate function for handling array:
`def get_array_typescript(param_name: Optional[str], param_dic: dict, depth: int = 0) -> str:`

so if x is an array of enum:
Json schema:
`"x": {"type": "array", "items": {"type": "string", "enum": ["option1", "option2", "option3"]}}
`Type script:
`x: ("option1"|"option2"|"option3")[]`

Reply:
You can take a look at: https://github.com/MeetKai/functionary/blob/train_lora/tests/test_schema.py
to see the output of converting json schema to typescript schema

Reply:
class A (Enum, str):
 option1 = "1"
 option2 = "2"
 items = List[A]